### PR TITLE
gravel: api/services: available space must account for reservations

### LINKED ISSUE #236: gravel: reservations API not providing raw reservation, just service reservation
Essentially we're not providing the caller with the raw reserved space (i.e., service reservation * replicas), which makes the UI (and thus the user) believe they should have a lot more space available than they actually do.

### PR DESCRIPTION
**Pull request by jecluis**
Resolves #236

Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

**Review by mgfritch — APPROVED** [review]

File: src/gravel/controllers/services.py [hunk 1]
```diff
@@ -110,6 +113,12 @@ def total_raw_reservation(self) -> int:
             total += (service.reservation * service.replicas)
         return total
 
+    @property
+    def available(self) -> int:
```
Comment (jhesketh) [inline src/gravel/controllers/services.py:117]: This property isn't very clearly named. The object is `Services` so in the context of thinking about services "available" does not make me think of available space.

**Review by jhesketh — COMMENTED** [review]
Just a nit around the naming of the property. The change otherwise looks fine.

(Another thought could be to have this as part of the Node object instead)

**Review by votdev — APPROVED** [review]
LGTM, but requires some adaption in the UI. Do you want to do that in this PR or shall i open a glass related PR?

```
--- a/src/glass/src/app/core/dashboard/widgets/capacity-dashboard-widget/capacity-dashboard-widget.component.ts	(revision b6fca28e7bf4328bb086a292fb6308f9c4a20600)
+++ b/src/glass/src/app/core/dashboard/widgets/capacity-dashboard-widget/capacity-dashboard-widget.component.ts	(date 1615545673762)
@@ -31,8 +31,8 @@
     // @ts-ignore
     this.subscription = this.loadDataEvent.subscribe(() => {
       this.chartData = [
-        { name: translate(TEXT('Assigned')), value: this.data.available - this.data.reserved },
-        { name: translate(TEXT('Unassigned')), value: this.data.reserved }
+        { name: translate(TEXT('Assigned')), value: this.data.reserved },
+        { name: translate(TEXT('Unassigned')), value: this.data.available }
       ];
     });
   }
```

**Comment by jecluis** [comment]
if you could do that in another PR I'd appreciate it immensely :)

**Pull request merged**